### PR TITLE
PR: Allow single dedent with empty line

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes
Allow unindenting with a single empty line.

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes https://github.com/spyder-ide/spyder/issues/10146

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

File: spyder/plugins/editor/widgets/codeeditor.py [hunk 1]
```diff
@@ -2739,8 +2740,13 @@ def fix_indent_smart(self, forward=True, comment_or_string=False):
 
         # TODO untangle this block
         if prevline and not bracket_stack and not prevtext.endswith(':'):
-            cur_indent = self.get_block_indentation(block_nb - 1)
-            is_blank = not self.get_text_line(block_nb - 1).strip()
+            above_block_nb = block_nb - 1
+            # If the block above is empty, check the previous one
+            if (ignore_empty_line and self.get_text_line(above_block_nb) == ''
```
Comment (impact27) [inline spyder/plugins/editor/widgets/codeeditor.py]: This could be a while to allow for any number of empty lines. I think a single empty line is better.

Comment (ccordoba12) [reply spyder/plugins/editor/widgets/codeeditor.py]: I think this is fine, i.e. pressing backspace dedents 4 spaces if the previous line is empty (as requested on issue #10146) but it goes two the first column after two or more empty lines.

@bcolsen, what do you think?

Comment (ccordoba12) [reply spyder/plugins/editor/widgets/codeeditor.py]: Now that removing trailing spaces only works when the option is turned on in our Preferences, I think we can merge this one as it is.

Comment (ccordoba12) [reply spyder/plugins/editor/widgets/codeeditor.py]: (That was done in PR #10261).

**Comment by ccordoba12** [comment]
Sorry, I think this is not really necessary after PR #10261.

**Pull request closed without merging**